mothernet/prediction/mothernet_additive.py

```diff
@@ -34,13 +34,14 @@ def extract_additive_model(model, X_train, y_train, config=None, device="cpu", i
         n_classes = 1 if regression else len(np.unique(y_train))
         n_features = X_train.shape[1]
 
-        ys = torch.Tensor(y_train).to(device).unsqueeze(1)
-        xs = torch.Tensor(X_train).to(device).unsqueeze(1)
+        if len(X_train.shape) == 2:
```

Comment:
hm I have to see why the tests are not being run on this, but this seems strange. you're not converting to tensor outside of the if branch.